### PR TITLE
Image-URL-helper: add option to exclude images from being promoted

### PR DESCRIPTION
**Pull request by Halamix2**
**Description**

Changes proposed in this pull request:

- Add `excludes-list` flag for `promote` command
- join old registry to the new one in `values.yaml` file so the structure reflects promoted images 
- fix map of inconsistent images

**Related issue(s)**

Blocked by #4479

### CODE REVIEW

File: development/image-url-helper/pkg/promote/promote.go [hunk 1]
```diff
@@ -103,6 +113,15 @@ func GetWalkFunc(ResourcesDirectoryClean, targetContainerRegistry, targetTag str
 	}
 }
 
+func isFileExcluded(ResourcesDirectoryClean, path string, excludes []string) bool {
```
Comment (dekiel) [inline development/image-url-helper/pkg/promote/promote.go:116]: Use map to hold excluded files. With list you have to iterate over whole list for every values.yaml. A map better fit to the usage of data structure.

File: development/image-url-helper/pkg/promote/promote.go [hunk 2]
```diff
@@ -224,3 +243,22 @@ func saveToFile(path string, lines []string) error {
 	}
 	return nil
 }
+
+func ParseExcludes(excludesListFilename string) ([]string, error) {
```
Comment (dekiel) [inline development/image-url-helper/pkg/promote/promote.go:247]: argument excludesListFilename is getting value directly from excludeslist cli argument. Cli argument has description it contains list of excluded images. I expect to pass image names to this cli argument, not path to file with list excluded files. Description of cli argument is missleading.

**Review by dekiel — CHANGES_REQUESTED** [review]

File: development/image-url-helper/pkg/promote/promote.go (hunk 2 above)
Comment (Halamix2) [reply development/image-url-helper/pkg/promote/promote.go]: I've updated the CLI flag description to `Path to the file containing a list of excluded images`

File: development/image-url-helper/pkg/promote/promote.go (hunk 1 above)
Comment (Halamix2) [reply development/image-url-helper/pkg/promote/promote.go]: excludes returned by the `ParseExcludes` function are now in a map

**Review by dekiel — APPROVED** [review]

**Pull request merged**